grpc-proxy/README.md

```diff
@@ -0,0 +1,22 @@
+### Steps to run this sample:
```

Comment:
I know we haven't been doing it in many of our other examples, but can we get an overview here on what this demonstrates? E.g. "This sample shows how to compress data via a custom gRPC proxy using a custom codec".

Reply:
Added a brief overview.